modules/kubernetes/ingress-nginx/main.tf

```diff
@@ -22,33 +22,100 @@ terraform {
 resource "kubernetes_namespace" "this" {
   metadata {
     labels = {
-      name                = join("-", compact(["ingress-nginx", var.name_override]))
+      name                = "ingress-nginx"
       "xkf.xenit.io/kind" = "platform"
     }
-    name = join("-", compact(["ingress-nginx", var.name_override]))
+    name = "ingress-nginx"
   }
 }
 
 resource "helm_release" "ingress_nginx" {
+  for_each = {
+    for s in ["ingress-nginx"] :
+    s => s
+    if var.public_private_enabled == false
+  }
+
   repository = "https://kubernetes.github.io/ingress-nginx"
   chart      = "ingress-nginx"
   name       = "ingress-nginx"
   namespace  = kubernetes_namespace.this.metadata[0].name
   version    = "3.35.0"
   values = [templatefile("${path.module}/templates/values.yaml.tpl", {
-    http_snippet           = var.http_snippet,
-    name_override          = var.name_override,
-    provider               = var.cloud_provider,
-    ingress_class          = join("-", compact(["nginx", var.name_override])),
-    internal_load_balancer = var.internal_load_balancer,
+    http_snippet           = var.http_snippet
+    provider               = var.cloud_provider
+    ingress_class          = "nginx"
+    internal_load_balancer = false
+    default_certificate = {
+      enabled         = var.default_certificate.enabled
+      dns_zone        = var.default_certificate.dns_zone
+      namespaced_name = "ingress-nginx/ingress-nginx"
+    }
+    extra_config          = var.extra_config
+    extra_headers         = var.extra_headers
+    linkerd_enabled       = var.linkerd_enabled
+    multiple_ingress      = false
+    default_ingress_class = false
```

Comment:
This should probably be set to true even if it the class is nginx.

Reply:
removed `multiple_ingress` variable